test/jdk/java/awt/event/ComponentEvent/ComponentEventTest.java

```diff
@@ -0,0 +1,395 @@
+                    + components[i].getClass());
+        }
+
+        // Disable the components and do the same set of operations
+
+        EventQueue.invokeAndWait(() -> {
+            currentComponent.setEnabled(false);
+            resetFrame();
+        });
+        robot.delay(DELAY);
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> {
+            currentComponent.setVisible(false);
+            resetFrame();
+        });
+
+        robot.delay(DELAY);
+        if (!componentHidden) {
+            throw new RuntimeException(
+                "FAIL: ComponentHidden not triggered for disabled "
+                    + components[i].getClass());
+        }
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> {
+            currentComponent.setVisible(false);
+            resetFrame();
+        });
+
+        robot.delay(DELAY);
+        if (componentHidden) {
+            throw new RuntimeException(
+                "FAIL: ComponentHidden triggered when setVisible(false) "
+                    + "called for a hidden disabled "
+                    + components[i].getClass());
+        }
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> {
+            currentComponent.setVisible(true);
+            resetFrame();
+        });
+
+        robot.delay(DELAY);
+        if (!componentShown) {
+            throw new RuntimeException(
+                "FAIL: ComponentShown not triggered for disabled "
+                    + components[i].getClass());
+        }
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> {
+            currentComponent.setVisible(true);
+            resetFrame();
+        });
+
+        robot.delay(DELAY);
+        if (componentShown) {
+            throw new RuntimeException(
+                "FAIL: ComponentShown triggered when setVisible(true) "
+                    + "called for a shown disabled "
+                    + components[i].getClass());
+        }
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> {
+            currentComponent.setLocation(currentComponent.getLocation().x - 1,
+                currentComponent.getLocation().y);
+            resetFrame();
+        });
+
+        robot.delay(DELAY);
+        if (!componentMoved) {
+            throw new RuntimeException(
+                "FAIL: ComponentMoved not triggered for disabled "
+                    + components[i].getClass());
+        }
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> {
+            currentComponent.setSize(currentComponent.getSize().width - 1,
+                currentComponent.getSize().height);
+            resetFrame();
+        });
+
+        robot.delay(DELAY);
+        if (!componentResized) {
+            throw new RuntimeException(
+                "FAIL: ComponentResized not triggered for disabled "
+                    + components[i].getClass());
+        }
+
+    }
+
+    private static void resetFrame() {
```

Comment:
```suggestion
    private static void revalidateFrame() {
```
`revalidateFrame` is a better name, it's more descriptive: you *invalidate* and then *validate* the frame.